src/utils.php

```diff
@@ -645,3 +649,133 @@ function ensure_dir( $dir ) {
 
 	return realpath( $dir );
 }
+
+/**
+ * Create a file named ".lastruntime" in the root directory and write the current Unix time to it.
+ *
+ * @return bool True if the file was successfully written, false otherwise.
+ */
+function write_last_runtime_to_file(): bool {
+	$filename = '.lastruntime';
+	// Get the current Unix time.
+	$current_unix_time = time();
+
+	// Define the path to the .lastruntime file in the root directory.
+	$file_path = root( "/{$filename}" );
+
+	// Write the current Unix time to the .lastruntime file.
+	$result = file_put_contents( $file_path, $current_unix_time );
+
+	// Check if the file was successfully written.
+	return $result !== false;
+}
+
+/**
+ * Check if the mtime of a directory has changed since the last check.
+ *
+ * @param string $dir The directory path to check.
+ *
+ * @return bool True if the mtime has changed, false otherwise.
+ */
+function should_skip_check_based_on_mtime( string $dir ): bool {
+	$lastruntime_file = root( '/.lastruntime' );
+
+	if ( file_exists( $lastruntime_file ) ) {
+		$last_runtime  = (int) file_get_contents( $lastruntime_file );
+		$current_mtime = filemtime( $dir );
+
+		if ( $current_mtime <= $last_runtime ) {
```

Comment:
This could fail unexpectedly.
The `filemtime` function will return `int|false`; the `file_get_contents` function will return `string|false`. If both return `false`, then `if(false <= false` will evaluate to `true`, and the check will be skipped.
In any utility code, like `slic` is, write defensive code.